google/cloud/bigtable/backup.py

```diff
@@ -0,0 +1,392 @@
+                             project ID on the Instance's client, or if the
+                             parsed instance ID does not match the Instance ID.
+        """
+        match = _BACKUP_NAME_RE.match(backup_pb.name)
+        if match is None:
+            raise ValueError(
+                "Backup protobuf name was not in the expected format.", backup_pb.name
+            )
+        if match.group("project") != instance._client.project:
+            raise ValueError(
+                "Project ID of the Backup does not match the Project ID "
+                "of the instance's client"
+            )
+
+        instance_id = match.group("instance_id")
+        if instance_id != instance.instance_id:
+            raise ValueError(
+                "Instance ID of the Backup does not match the Instance ID "
+                "of the instance"
+            )
+        backup_id = match.group("backup_id")
+        cluster_id = match.group("cluster_id")
+
+        match = re.compile(
+            r"^projects/(?P<project>[^/]+)/"
+            r"instances/(?P<instance_id>[a-z][-a-z0-9]*)/"
+            r"tables/(?P<table_id>[_a-zA-Z0-9][-_.a-zA-Z0-9]*)$"
+        ).match(backup_pb.source_table)
+        table_id = match.group("table_id") if match else None
+
+        expire_time = backup_pb.expire_time
+
+        backup = cls(
+            backup_id,
+            instance,
+            cluster_id=cluster_id,
+            table_id=table_id,
+            expire_time=expire_time,
+        )
+        backup._start_time = backup_pb.start_time
+        backup._end_time = backup_pb.end_time
+        backup._size_bytes = backup_pb.size_bytes
+        backup._state = backup_pb.state
+
+        return backup
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+        return other.backup_id == self.backup_id and other._instance == self._instance
+
+    def __ne__(self, other):
+        return not self == other
+
+    def create(self, cluster_id=None):
+        """Creates this backup within its instance.
+
+        :type cluster_id: str
+        :param cluster_id: (Optional) The ID of the Cluster for the newly
+                           created Backup.
+
+        :rtype: :class:`~google.api_core.operation.Operation`
+        :returns: :class:`~google.cloud.bigtable_admin_v2.types._OperationFuture`
+                  instance, to be used to poll the status of the 'create' request
+        :raises Conflict: if the Backup already exists
+        :raises NotFound: if the Instance owning the Backup does not exist
+        :raises BadRequest: if the `table` or `expire_time` values are invalid,
+                            or `expire_time` is not set
+        """
+        if not self._expire_time:
+            raise ValueError('"expire_time" parameter must be set')
+            # TODO: Consider implementing a method that sets a default value of
```

Comment:
Can we create an issue to track this as a feature request?

Reply:
Certainly, that's the idea. I'd just wait until the initial release is merged, but please let know if you'd rather see it sooner.